app/src/main/java/com/example/composetemplate/managers/FirebaseNetworkManager.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.composetemplate.managers
+
+import com.example.composetemplate.data.remote.base.BaseNetworking
+import com.example.composetemplate.data.remote.base.BaseRequest
+
+class FirebaseNetworkManager (
```

Comment:
Now that I’m reviewing this alone, I don't understand why you added the FirebaseNetworkManager. You should be using the NetworkManager (or MainNetworkManager as you renamed it), and that's all. The NetworkManager accepts a BaseNetworking (or its subclass) in its constructor. The correct approach is to create a FirebaseNetworking class (as a subclass of BaseNetworking), inject it into NetworkManager, and implement the necessary functionality.

Reply:
abstract class  A : BaseNetworking {
    
}

example for what we talk about when I get there